core/src/utils/content/index.ts

```diff
@@ -0,0 +1,130 @@
+  if (isScrollable && node.scrollHeight >= node.clientHeight) {
+    return true;
+  }
+  return false;
+}
+
+const getScrollContainer = (node: Element) => {
+  if (hasOverflowScroll(node)) {
+    return node;
+  }
+  for (const item of Array.from(node.children)) {
+    if (item && hasOverflowScroll(item)) {
+      return item;
+    }
+  }
+  return null;
+}
+
+/**
+ * Waits for the element host fully initialize before
+ * returning the inner scroll element.
+ *
+ * For `ion-content` the scroll target will be the result
+ * of the `getScrollElement` function.
+ *
+ * For custom implementations it will be the element host
+ * or a selector within the host, if supplied through `scrollTarget`.
+ */
+export const getScrollElement = async (el: Element) => {
+  await new Promise(resolve => componentOnReady(el, resolve));
+
+  if (el?.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    return (el as HTMLIonContentElement).getScrollElement();
+  }
+
+  const scrollContainer = getScrollContainer(el) as HTMLElement;
+
+  if (scrollContainer) {
+    return scrollContainer;
+  }
+
+  return el as HTMLElement;
+}
+
+/**
+ * Queries the element matching the selector for IonContent.
+ * See ION_CONTENT_SELECTOR for the selector used.
+ */
+export const findIonContent = (el: Element) => {
+  /**
+   * First we try to query the custom scroll host selector in cases where
+   * the implementation is using an outer `ion-content` with an inner custom
+   * scroll container.
+   */
+  const customContentHost = el.querySelector<HTMLElement>(ION_CONTENT_CLASS_SELECTOR);
+  if (customContentHost) {
+    return customContentHost;
+  }
+  return el.querySelector<HTMLElement>(ION_CONTENT_SELECTOR);
+}
+
+/**
+ * Queries the closest element matching the selector for IonContent.
+ */
+export const findClosestIonContent = (el: Element) => {
+  return el.closest<HTMLElement>(ION_CONTENT_SELECTOR);
+}
+
+/**
+ * Scrolls to the top of the element. If an `ion-content` is found, it will scroll
+ * using the public API `scrollToTop` with a duration.
+ */
+export const scrollToTop = (el: HTMLElement, durationMs: number): Promise<any> => {
+  if (el.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    const content = el as HTMLIonContentElement;
+    return content.scrollToTop(durationMs);
+  }
+  return Promise.resolve(el.scrollTo({
+    top: 0,
+    left: 0,
+    behavior: durationMs > 0 ? 'smooth' : 'auto'
+  }));
+}
+
+/**
+ * Scrolls by a specified X/Y distance in the component. If an `ion-content` is found, it will scroll
+ * using the public API `scrollByPoint` with a duration.
+ */
+export const scrollByPoint = (el: HTMLElement, x: number, y: number, durationMs: number) => {
+  if (el.tagName.toLowerCase() === ION_CONTENT_ELEMENT_SELECTOR) {
+    const content = el as HTMLIonContentElement;
+    return content.scrollByPoint(x, y, durationMs);
+  }
+  return Promise.resolve(el.scrollTo({
```

Comment:
Can we use [Element.scrollBy](https://developer.mozilla.org/en-US/docs/Web/API/Element/scrollBy) instead?